### ISSUE TITLE
Add tiling support for channels (not MDI)

### ISSUE THREAD

**Issue by AlexandreBonneau** [kvirc/KVIrc#2354]
## The problem
This [commit](https://github.com/kvirc/KVIrc/commit/a8b12f44d0e747d143b61c50ed4f8c580045a259) was a dark day for me, as I extensively use 'tiling' in order to be able to check many channels in one glance.
The setups I'm using with N channels displayed:
- 8 usually, in 3 columns with 2, 3 and 3 channels in each, or
- 9 in a 3x3 fashion, or
- 10 in 3 columns with 4, 3 and 3 channels in each
- etc.

Being able to follow multiple channels at the same time is just *impossible* with the current Kvirc version without having to click on each channel names and manually do some 'polling' to check if somebody answered to you without highlighting your nickname. And even if you got highlighted, you *still* need to do an action, a click usually, to be able to read that answer.

## The reality
However we *all* do **multitasking** and it's pretty usual to have parallel conversations; for instance when :
- asking a question in a channel #d3.js where the usual response time is many hours,
- checking #javascript from time to time to give help if able,
- talking in ##linux about the weird error you just saw in the syslog while setting up your apache server,
- ...while talking in #apache about how to configure a particular option
- etc.

In that case, you want to keep an eye on each of those chans to be able to react immediately, without having to 1) poll each channel regularly, 2) scan each channel for new contents from the last seen line kvirc displays.

## The solution
**Tiling** is the easy and logical answer to multitasking; it would be great if that feature would come back into Kvirc, because being stuck on `v4.2.0` is no fun!

Why is it no fun? Because MDI is broken and not really *easy* to use.

&#x1F534; **When I'm asking for a tiling feature, this is *not* asking for MDI back.** &#x1F534;

### MDI sucks (kinda)
Currently with `4.2`, you *can* tile multiple channels side by side, however *this has quite a lot of drawbacks*:
- you can (accidentally or not) move one of the channel by dragging its titlebar,
- some additional channel/windows can be automatically opened (private messages, `*status` and `*route_replies` windows when using `znc`, etc.)
- when you reconnect to a server, all its child channels are focused one after the other
...and all those actions mess with the tiling alignment, the order, or both.

When that happens, you have to do the following to get back to the state you were in before, in that exact order:
1. Minimize any opened window that you do not want to see in the final tiling
2. Select *each* window in **reverse order** of importance (ie. clicking 9 times if you have a 3x3 tile structure); *(this is mandatory if you want the tiling method to order the window in the same order than before)*
3. Select the tiling method by selecting `Window > Tiling method > 'Grille complète d'Anodine'` (Sorry I'm not sure about the English translation for this one)

This is quickly annoying when you have to do that many times a day or whenever you want to open temporary channels.

Finally, the 'Automatic tiling' method is a mess cognitive-wise: the channels order changes pretty often and you cannot set the channels order definitively to have them sorted by importance/topic.

### The feature request
- The console windows should not be treated like channels (displaying the console could use all the workspace like today)
- Private message windows should not be treated like regular channels (I'm not sure how those could be displayed without messing the channel tiling)
- When a server reconnect happens, the console and the child channels should not be focused
- The user should be able to set *once* the order in which each displayed channel should be organized, and the tiling method should remember and use that
- The user should be able to define how many columns they want, as well as how many windows for each of those columns too
- The user should be able to define how the ordering should be done, either from top left to bottom right by going vertically, or going horizontally
- No need for a window decoration for each channel. Perhaps add a way to set the window order directly with an input accepting the order index

tl;dr: Add a tiling feature where you could set the channels order and layout type, so that you would always know where are situated each channel, and see each one in a single glance

**Comment by JeroenvIS** [comment]
As I usually run Ubuntu LTS versions for a long time, I only ran into this recently by upgrading to Bionic / 18.04 - I too loved the multitasking made possible by having a lot of channel windows in view simultaneously, in KVIrc 4.2.0 - which is not possible anymore with current KVIrc versions.

Personally I even accept all the minor issues/bugs that came from the MDI approach, as at least with that I could arrange the windows exactly the way I wanted it.

So I second this feature request... And as long as it's not implemented, will seriously look for other IRC clients and/or find another way to run an Equilibrium build again.

Is there any chance that a tiling feature like this one is added back? Or should I just switch to WeeChat or irssi? Even though they are text based, at least they appear to be flexible with splitting channel windows and have loads of other features too...

**Comment by DarthGandalf** [comment]
The current workaround for the absence of MDI is to use your window manager by "undocking" channel windows, so that they become their own windows managed by WM. Those windows can be tiled however you want, depending on the particular WM used.
